### PR TITLE
Create a Global error Handler for a Front end project

### PR DESCRIPTION
**Pull request by sunitaguptaEviden**
Create a Global error Handler for Front end project

### CODE REVIEW

File: src/EPR.Calculator.Frontend/Middleware/GlobalErrorHandlerMiddleware.cs
[the remark this answers is not in the document]
Comment (sunitaguptaEviden) [reply src/EPR.Calculator.Frontend/Middleware/GlobalErrorHandlerMiddleware.cs:1]: @Mazar-Shaik Implementation of the .Net8 IExceptionHandler Interface is Done

File: src/EPR.Calculator.Frontend/Program.cs [hunk 1]
```diff
@@ -36,6 +39,10 @@
     // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
     app.UseHsts();
 }
+else
+{
+    app.UseExceptionHandler(_ => { });
```
Comment (udaydefra) [inline src/EPR.Calculator.Frontend/Program.cs:44]: Please remove blank arrow function

Comment (sunitaguptaEviden) [reply src/EPR.Calculator.Frontend/Program.cs]: @udaydefra , For adding the ExceptionHandlerMiddleware to the request pipeline, blank function was giving error but with using error page path the adding is successful

File: src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs [hunk 2]
```diff
@@ -24,7 +24,7 @@ public IActionResult RunCalculator(InitiateCalculatorRunModel calculationRunMode
                 return this.View(CalculationRunNameIndexView);
             }
 
-            this.HttpContext.Session.SetString(SessionConstants.CalculationName, calculationRunModel.CalculationName);
+            this.HttpContext.Session.SetString(SessionConstants.CalculationName, calculationRunModel.CalculationName ?? string.Empty);
```
Comment (udaydefra) [inline src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs:27]: Please do the changes as suggested

Comment (sunitaguptaEviden) [reply src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs]: @udaydefra I have fixed the suggestion

**Review by udaydefra — APPROVED** [review]
@sunitaguptaEviden LGTM. Thanks a lot for changes

File: src/EPR.Calculator.Frontend/Program.cs [hunk 3]
```diff
@@ -26,13 +27,15 @@
...
 builder.Services.AddHttpClient();
 
+builder.Services.AddExceptionHandler<GlobalExceptionHandler>();
+
 var app = builder.Build();
 
 // Configure the HTTP request pipeline.
+app.UseExceptionHandler("/StandardError/Index");
+
 if (!app.Environment.IsDevelopment())
```
Comment (Mazar-Shaik) [inline src/EPR.Calculator.Frontend/Program.cs:37]: what in case of production?

**Review by Mazar-Shaik — APPROVED** [review]

File: src/EPR.Calculator.Frontend/Middleware/GlobalErrorHandlerMiddleware.cs
[the remark this answers is not in the document]
Comment (Mazar-Shaik) [reply src/EPR.Calculator.Frontend/Middleware/GlobalErrorHandlerMiddleware.cs:1]: Thanks @sunitaguptaEviden for implementing. Looks good now

**Pull request merged**